### PR TITLE
Mappings validation

### PR DESCRIPTION
**Pull request by yasuba**

### CODE REVIEW

File: modules/core/src/main/scala/mapping.scala [hunk 1]
```diff
@@ -17,6 +17,26 @@ trait Mapping[F[_]] {
...
+    def typeMappingFieldNames: List[String] = typeMappings.collect {
+      case om: ObjectMapping => om.fieldMappings.flatMap {
+        case _: CursorField[_] => Nil
+        case _: CursorAttribute[_] => Nil
+        case fm => List(fm.fieldName)
+      }
+    }.flatten
+
+    val mappingTypesExistInSchema: Boolean =
+      typeMappingNames.forall(name => schema.types.map(_.name).contains(name))
```
Comment (yasuba) [inline modules/core/src/main/scala/mapping.scala:32]: It would be good to have some kind of logging or feedback about which types and fields are missing.

Comment (milessabin) [reply modules/core/src/main/scala/mapping.scala:32]: Maybe have `validate` return a list of errors encoded as json? See `QueryInterpreter.mkOneError`.

**Review by milessabin — COMMENTED** [review]
Just that one comment, and needs rebasing, but otherwise LGTM!

File: modules/core/src/main/scala/mapping.scala (hunk 1 above)
Comment (yasuba) [reply modules/core/src/main/scala/mapping.scala:32]: Sounds good to me.

File: modules/core/src/main/scala/mapping.scala [hunk 2]
```diff
@@ -5,21 +5,43 @@ package edu.gemini.grackle
...
   implicit val M: Monad[F]
 
   val schema: Schema
   val typeMappings: List[TypeMapping]
-
+  
   def typeMapping(tpe: Type): List[TypeMapping] =
     typeMappings.filter(_.tpe.nominal_=:=(tpe))
 
+  def validate: Either[NonEmptyChain[Json], String] = {
```
Comment (milessabin) [inline modules/core/src/main/scala/mapping.scala:23]: Ahh ... right, I see what you means about the result type looking odd. Why not return `Chain[Json]` where an empty chain indicates no errors detected?

**Review by milessabin — COMMENTED** [review]
Aside from the signature tweak, LGTM :-)

File: modules/core/src/main/scala/mapping.scala (hunk 2 above)
Comment (yasuba) [reply modules/core/src/main/scala/mapping.scala:23]: Ok, yes that sounds good.

**Review by milessabin — APPROVED** [review]
LGTM!

**Pull request merged**